packages/core/src/Modules/Cashier/Containers/p2p-cashier.jsx

```diff
@@ -1,95 +1,76 @@
-        const current_query_param = queryString.parse(this.props.location.search);
-        delete current_query_param.order;
+const P2PCashier = ({ currency, history, is_mobile, is_virtual, local_currency_config, location, residence }) => {
+    const [order_id, setOrderId] = React.useState(null);
 
-        const order_id_param = input_order_id
-            ? { ...current_query_param, order: input_order_id }
-            : { ...current_query_param };
+    React.useEffect(() => {
+        const url_params = new URLSearchParams(location.search);
+        if (url_params.has('order')) {
+            setQueryOrder(url_params.get('order'));
+        }
+    }, []);
 
-        if (this.state.order_id !== input_order_id) {
-            // Changing query params
-            history.push({
-                pathname: '/cashier/p2p',
-                search: queryString.stringify(order_id_param),
+    const setQueryOrder = input_order_id => {
+        const pathname = '/cashier/p2p';
+        if (input_order_id === null) {
+            history.replace({
+                pathname,
+                hash: location.hash,
+            });
+        } else {
+            const url_params = new URLSearchParams(location.search);
+            if (order_id !== input_order_id) {
+                url_params.set('order', input_order_id);
+            }
+            history.replace({
+                pathname,
+                search: `?${url_params.toString()}`,
                 hash: location.hash,
             });
-            this.setState({ order_id: input_order_id });
         }
-    };
 
-    componentDidMount() {
-        const url_params = new URLSearchParams(this.props.location.search);
-        const order_id = url_params.get('order');
+        setOrderId(input_order_id);
+    };
 
-        if (order_id) {
-            this.setQueryOrder(order_id);
-        }
-    }
-    render() {
-        const {
-            currency,
-            local_currency_config,
-            loginid,
-            is_virtual,
-            residence,
-            setNotificationCount,
-            location,
-            is_mobile,
-        } = this.props;
-        const { order_id } = this.state;
-        return (
-            <P2P
-                should_show_verification={/verification/.test(location.hash)}
-                websocket_api={WS}
-                lang={getLanguage()}
-                client={{ currency, local_currency_config, is_virtual, residence, loginid }}
-                server_time={ServerTime}
-                setNotificationCount={setNotificationCount}
-                order_id={order_id}
-                setOrderId={this.setQueryOrder}
-                is_mobile={is_mobile}
-                poi_url={routes.proof_of_identity}
-            />
-        );
-    }
-}
+    return (
+        <P2P
+            LocalStorage={P2pStorage}
+            client={{ currency, local_currency_config, is_virtual, residence }}
+            is_mobile={is_mobile}
```

Comment:
`isMobile` utility function is available in `@deriv/shared`. Can it be used directly in p2p package instead of passing it as prop?